Digipost/OAuthToken+IDToken.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  OAuthToken+IDToken.swift
+//  Digipost
+//
+//  Created by Håkon Bogen on 19/05/15.
+//  Copyright (c) 2015 Posten. All rights reserved.
+//
+
+import UIKit
+
+private struct OAuthTokenIdTokenConstants {
+    static let nonce = "nonce"
+    static let aud = "aud"
+}
+
+extension OAuthToken {
+
+    class func isIdTokenValid(idToken : String?, nonce : String) -> Bool {
+        if let actualIDToken = idToken {
+            let idTokenContentArray  = idToken?.componentsSeparatedByString(".")
+            if idTokenContentArray?.count == 2 {
+                if let base64EncodedJson = idTokenContentArray?[1] {
+                    var numberOfCharactersAdded = 0
+                    var error : NSError?
+                    var alteredBase64EncodedJson = base64EncodedJson
+                    var base64Data : NSData?
+                    while base64Data == nil {
```

Comment:
Probably overkill yes, might there have been an issue with the Sporing client that needed to add that. If there is no need we can just strip the trailing, if thats valid base64 in our case